lib/diagnostic.js

```diff
@@ -32,14 +52,20 @@ const newArray = function newArray(/* length, defaultValue */) {
 // to the result of invoking defaultValue with the
 // element's index as an argument
 
+
 // Questions 5
 // Write a function that adds or replaces a method on an object
 // using the given propertyName
 // If the argument obj is unset, create a new object
 // The function you write should return
 // the object passed in as an argument or the object created
-const addMethod = function addMethod(/* propertyName, method, obj */) {
-
+const addMethod = function addMethod(property, method, obj)
+{
+  let affObj = obj || {};
+  affObj.property = function(affObj)
+  {
+    return(affObj);
+  };
 };
```

Comment:
I think you're on the right track here, but again, the details are off. Notice that you are taking in 3 arguments. Are all 3 used in your code? Take a look at this solution:
```
const addMethod = function addMethod(propertyName, method, obj) {
  if (!obj) {
    obj = {};
  }
  obj[propertyName] = method;
  return obj;
};
```